superset/models/dashboard.py

```diff
@@ -171,8 +172,21 @@ def url(self) -> str:
 
     @property
     def datasources(self) -> Set[BaseDatasource]:
-        # pylint: disable=using-constant-test
-        return {slc.datasource for slc in self.slices if slc.datasource}
```

Comment:
@villebro per your point we could also use [`functools.lru_cache`](https://docs.python.org/3/library/functools.html#functools.lru_cache) to cache the `datasource` property for the lifetime of the request. It's likely safer than using `@cache.memoize(...)` as there's likely little concern that the data will be stale (within the context of a request).